client/lib/create-selector/index.js

```diff
@@ -31,14 +31,13 @@ const VALID_ARG_TYPES = [ 'number', 'boolean', 'string' ];
 const DEFAULT_GET_DEPENDANTS = state => state;
 
 /**
- * At runtime, assigns a function which returns a cache key for the memoized
- * selector function, given a state object and a variable set of arguments. In
- * development mode, this warns when the memoized selector is passed a complex
+ * In development mode, this warns when the memoized selector is passed a complex
  * object argument, as these cannot be depended upon as reliable cache keys.
+ * Note that the underlying function is created at runtime depending on the environment
  *
  * @type {Function} Function returning cache key for memoized selector
  */
-const DEFAULT_GET_CACHE_KEY = ( () => {
+const warnAgainstComplexArguments = ( () => {
```

Comment:
Reason for decoupling: I plan to use getCacheKey in the body of the selector in order to bust the cache for that key only. Without decoupling the warning would show multiple times (and breaks tests too)